engine/src/operators/OrderBy.cpp

```diff
@@ -214,7 +214,34 @@ std::unique_ptr<ral::frame::BlazingTable> generate_partition_plan(const std::vec
 	}
 
 	int num_nodes = context->getTotalNodes();
-	cudf::size_type total_num_partitions = (double)table_num_rows*(double)avg_bytes_per_row/(double)num_bytes_per_order_by_partition;
+
+	// This mean master node does not have data, so we want to recompute the `avg_bytes_per_row` value
+	// using the average of all the samples that contains data
+	if (avg_bytes_per_row == 1 && samples.size() > 1 && table_num_rows > 0) {
```

Comment:
I understand what we are doing here, but this means we are now sending samples that have all the columns, but we ONLY need the columns, in case we need to estimate avg_bytes_per_row.

I think what we want to do instead is only send the columns we need, but add as extra metadata the avg_bytes_per_row for every sample. This will be able to be done more easily when we have the new communications.

I think we should postpone this PR until we have the new communications and implement it in such a way so that we are not sending those extra columns with the samples

Reply:
@Christian8491 now the the new communications has been merged in, could you modify this PR so that when you send the samples you also send an extra metadata that is AVG_BYTES_PER_ROW. This would be a new metadata tag, but it should be fairly straight forward to implement. Please let me know if you need help with this